### PR TITLE
fix(TDQ-15202): No DQ item from reference proejct shows when click Merge reference project button in main proejct

### PR DESCRIPTION
**Pull request by zshen-talend**

### CODE REVIEW

File: main/plugins/org.talend.cwm.management/src/main/java/org/talend/dq/nodes/DQRepositoryNode.java [hunk 1]
```diff
@@ -379,4 +382,12 @@ public IImage getIcon() {
     public Image getImage() {
         return null;
     }
+
+    @Override
+    public boolean equals(Object obj) {
+        if (super.equals(obj)) {
+            return this.getProject().equals(((DQRepositoryNode) obj).getProject());
```
Comment (xqliu-talend) [inline main/plugins/org.talend.cwm.management/src/main/java/org/talend/dq/nodes/DQRepositoryNode.java]: why call super.equals(obj) first ?

File: main/plugins/org.talend.cwm.management/src/main/java/org/talend/dq/nodes/SysIndicatorFolderRepNode.java [hunk 2]
```diff
@@ -156,13 +149,16 @@ private void createChildrenNode(boolean withDeleted, Project project) throws Per
...
+        boolean ignoreFolder = super.isIgnoreFolder(withDeleted, project, folder);
+        if (ignoreFolder) {
             return true;
         }
 
         // TDQ-11110: hidden some system indicator folders.
-        boolean isBusinessRules = folder.getLabel().equals(EResourceConstant.SYSTEM_INDICATORS_BUSINESS_RULES.getName());
-        boolean isPhoneNumber = folder.getLabel().equals(EResourceConstant.SYSTEM_INDICATORS_PHONENUMBER_STATISTICS.getName());
+        boolean isBusinessRules =
+                folder.getLabel().equals(EResourceConstant.SYSTEM_INDICATORS_BUSINESS_RULES.getName());
```
Comment (xqliu-talend) [inline main/plugins/org.talend.cwm.management/src/main/java/org/talend/dq/nodes/SysIndicatorFolderRepNode.java]: is it better: EResourceConstant.SYSTEM_INDICATORS_BUSINESS_RULES.getName(). equals(folder.getLabel()) ?

File: main/plugins/org.talend.cwm.management/src/main/java/org/talend/dq/nodes/SysIndicatorFolderRepNode.java [hunk 3]
```diff
@@ -156,13 +149,16 @@ private void createChildrenNode(boolean withDeleted, Project project) throws Per
...
+        boolean isPhoneNumber =
+                folder.getLabel().equals(EResourceConstant.SYSTEM_INDICATORS_PHONENUMBER_STATISTICS.getName());
```
Comment (xqliu-talend) [inline main/plugins/org.talend.cwm.management/src/main/java/org/talend/dq/nodes/SysIndicatorFolderRepNode.java]: is it better: EResourceConstant. SYSTEM_INDICATORS_PHONENUMBER_STATISTICS.getName(). equals(folder.getLabel()) ?

File: main/plugins/org.talend.cwm.management/src/main/java/org/talend/dq/nodes/DQRepositoryNode.java (hunk 1 above)
Comment (zshen-talend) [reply main/plugins/org.talend.cwm.management/src/main/java/org/talend/dq/nodes/DQRepositoryNode.java]: added commont to example what we did

File: main/plugins/org.talend.cwm.management/src/main/java/org/talend/dq/nodes/SysIndicatorFolderRepNode.java (hunk 2 above)
Comment (zshen-talend) [reply main/plugins/org.talend.cwm.management/src/main/java/org/talend/dq/nodes/SysIndicatorFolderRepNode.java]: done it

File: main/plugins/org.talend.cwm.management/src/main/java/org/talend/dq/nodes/SysIndicatorFolderRepNode.java (hunk 3 above)
Comment (zshen-talend) [reply main/plugins/org.talend.cwm.management/src/main/java/org/talend/dq/nodes/SysIndicatorFolderRepNode.java]: done it

**Pull request merged**